### PR TITLE
[API] Implement choosing province during checkout

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: features/addressing/managing_countries/adding_country.feature [hunk 1]
```diff
@@ -7,7 +7,8 @@ Feature: Adding a new country
     Background:
         Given I am logged in as an administrator
 
-    @ui @api
+#    @ui
```
Comment (Tomanhez) [inline features/addressing/managing_countries/adding_country.feature:10]:
```suggestion
#    @ui
```
Its needed?

File: src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php [hunk 2]
```diff
@@ -737,11 +758,47 @@ public function iShouldBeInformedThatCartIsNoLongerAvailable(): void
...
+    {
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+        Assert::same($response->getStatusCode(), 400);
+        Assert::true($this->isViolationWithMessageInResponse(
+            $response,
+            'Please select proper province.',
+            'billingAddress'
+        ));
+    }
```
Comment (Tomanhez) [inline src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:791]:
```suggestion
    /**
     * @Then /^I should not be able to specify province name manually for (billing|shipping) address$/
     */
    public function iShouldNotBeAbleToSpecifyProvinceNameManuallyForAddress(string addressType): void
    {
        /** @var Response $response */
        $response = $this->client->getResponse();
        Assert::same($response->getStatusCode(), 400);
        Assert::true($this->isViolationWithMessageInResponse(
            $response,
            'Please select proper province.',
            $addressType . 'Address'
        ));
    }
```
You can merge this two steps to one

Comment (GSadee) [reply src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php:791]: Not completely, because of line 766, I could extract asserts to a private method

**Review by AdamKasp — APPROVED** [review]

**Review by Tomanhez — APPROVED** [review]

**Comment by Tomanhez** [comment]
Thanks, Grzegorz! :tada:

**Pull request merged**